packages/components/src/InputFile/InputFile.tsx

```
@@ -469,10 +510,30 @@ function getFileUpload(
   file: File,
   key: string,
   uploadUrl?: string,
+  transformFilename?: (filename: string) => string,
 ): FileUpload {
+  let fileName = file.name;
+
+  if (transformFilename) {
+    try {
+      const transformed = transformFilename(file.name);
+      // Use transformed name if it's not empty, otherwise fall back to original
+      fileName = transformed && transformed.trim() ? transformed : file.name;
+    } catch (error) {
+      // If transformation fails, silently fall back to original filename
+      fileName = file.name;
```

@nad182 I don't know the right answer off the top of my head, but silently falling back to the original could cause a lot of problems for our users. Can we think of a better alternative?

---

Even, resorting to the default below is better than silently returning to the original one.

---

We can sent some error back to the consumer for them to decide what to do with it. I pushed another commit addressing this: https://github.com/GetJobber/atlantis/pull/2591/commits/c77d5e82bdbd4d20a96721796aee660c44e5c2f9

---

I don't love that approach... Let me look at how this component and the props we're proposing would actually get used in reality. Want to think through what the best realistic user experience would be for this.